packaging/docker/Dockerfile

```diff
@@ -0,0 +1,57 @@
+FROM centos:7
+LABEL maintainer="tdockendorf@osc.edu; johrstrom@osc.edu"
+
+# setup the ondemand repositories
+COPY files/ondemand-web.repo /etc/yum.repos.d/ondemand-web.repo
+COPY files/ondemand-ci-web.repo /etc/yum.repos.d/ondemand-ci-web.repo
+RUN curl https://yum.osc.edu/ondemand/RPM-GPG-KEY-ondemand -o /etc/pki/rpm-gpg/RPM-GPG-KEY-ondemand
+RUN rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-ondemand
+
+# update
+RUN yum install -y centos-release-scl 
+RUN yum -y update
+
+# install all the dependencies
+RUN yum install -y \
+        file \
+        lsof \
+        sqlite-devel \
+        sudo \
+        httpd24 \
+        rh-ruby25 \
+        rh-nodejs10 \
+        ondemand-passenger-6.0.4-2.el7.x86_64 \
+        ondemand-nginx-1.17.3-2.p6.0.4.el7.x86_64
+
+RUN yum clean all && rm -rf /var/cache/yum/*
+RUN rm -rf /var/cache/yum/*
+
+# copy basic stuff to /opt/ood and /var/www/ood/apps
+RUN mkdir -p /opt/ood
+RUN mkdir -p /var/www/ood/{apps,public,discover}
+RUN mkdir -p /var/www/ood/apps/{sys,dev,usr}
+
+COPY ../../mod_ood_proxy          /opt/ood/mod_ood_proxy
+COPY ../../nginx_stage/           /opt/ood/nginx_stage
+COPY ../../ood-portal-generator   /opt/ood/ood-portal-generator
+COPY ../../ood_auth_map           /opt/ood/ood_auth_map
+COPY ../../build/apps             /var/www/ood/apps/sys
```

Comment:
This won't work with docker build. You can't put things ouside Dockerfile directory into the container. Has this actually worked when tested?

Reply:
```
$ docker build -t test .
Sending build context to Docker daemon  459.3kB
Step 1/29 : FROM centos:7
 ---> 67fa590cfc1c
Step 2/29 : LABEL maintainer="tdockendorf@osc.edu; johrstrom@osc.edu"
 ---> Running in 16b96eac610a
Removing intermediate container 16b96eac610a
 ---> 84b39aa39f98
Step 3/29 : COPY ../../mod_ood_proxy          /opt/ood/mod_ood_proxy
COPY failed: Forbidden path outside the build context: ../../mod_ood_proxy ()
```

Reply:
Lol that's silly, I'd been using `buildah bud` that uses buildah to build a Dockerfile.

Reply:
Ya I think that project is one I've seen mentioned on Docker issue tracker because of people's frustration with this "feature" in Docker image building. I have resorted to ugly hacks of putting files on web server and using curl to avoid the layer getting big from local files.

Reply:
I am trying to run "docker build -t test .` at root of this project with simple Dockerfile and it's crashing my laptop. I think something with Docker is trying to pull in the entire project into a layer since build directory ends up getting stuffed into first layer or something equally as annoying.

Reply:
Ugh. Having a `build` directory from previous builds is going to blow up docker if Dockerfile is at root of project.  Once I did `rm -rf build/*` I was able to build simple container:

```
$ docker build -t test .
Sending build context to Docker daemon  13.79MB
Step 1/4 : FROM centos:7
 ---> 67fa590cfc1c
Step 2/4 : LABEL maintainer="tdockendorf@osc.edu; johrstrom@osc.edu"
 ---> Using cache
 ---> 84b39aa39f98
Step 3/4 : RUN mkdir -p /opt/ood
 ---> Using cache
 ---> dc14b8aac7a4
Step 4/4 : COPY mod_ood_proxy          /opt/ood/mod_ood_proxy
 ---> 41bdefbf2d2a
Successfully built 41bdefbf2d2a
Successfully tagged test:latest
```

Reply:
Yea it sends the entire current directory? Idk, I build images with `buildah`, so I hadn't run into it until now. The build directory alone seems to be 500 MB. 

`Sending build context to Docker daemon  776.6MB`

In any case, I solved this particular problem in the last commit.